File: ArmoryLib/src/main/java/com/domker/armory/framework/DexPath.java
package com.domker.armory.framework;

import static com.domker.armory.log.Logger.log;

/**
 * 对于dex的信息打印
 * <p>
 * Created by wanlipeng on 2019/4/8 5:14 PM
 */
public final class DexPath {

    /**
     * 打印当年运行时，系统的DexPathList里面包含的元素
     */
    public static void printDexPathList(ClassLoader classLoader) {
        try {
            Object dexPathList = Reflect.getFieldNoException(classLoader, "pathList");
            Object[] dexElements = Reflect.getFieldNoException(dexPathList, "dexElements");
            log("Print DexPathList Size is " + dexElements.length);
            for (int i = 0; i < dexElements.length; i++) {
                log("[%d] %s", i, dexElements[i]);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: ArmoryLib/src/main/java/com/domker/armory/framework/DexPath.kt
package com.domker.armory.framework

import com.domker.armory.framework.Reflect.findField
import com.domker.armory.log.log
import java.lang.reflect.Field

/**
 * Created by wanlipeng on 2017/11/12.
 */
object DexPath {

    /**
     * 打印当年运行时，系统的DexPathList里面包含的元素
     */
    fun printDexPathList(classLoader: ClassLoader) {
        val pathListField: Field
        try {
            pathListField = findField(classLoader, "pathList")
            val dexPathList = pathListField.get(classLoader)
            val dexElementsField = findField(dexPathList, "dexElements")
            val dexElements = dexElementsField.get(dexPathList) as Array<Any>
            log("Print DexPathList Size=" + dexElements.size)
            var i = 0
            dexElements.forEach { log("[${i++}] " + it) }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }


}